File: 3158. Find the XOR of Numbers Which Appear Twice.py
class Solution(object):
    def duplicateNumbersXOR(self, nums):
        dic = {}
        ans = 0
        for n in nums:
            if n in dic:
                dic[n]+=1
            else:
                dic[n] = 1
        for num, count in dic.items():
            if count == 2:
                ans ^= num
        return ans
        

File: 3158. Find the XOR of Numbers Which Appear Twice.cpp
class Solution {
public:
    int duplicateNumbersXOR(vector<int>& nums) {
        unordered_map<int, int> map;
        int ans = 0;
        for(int num : nums)
            map[num]++;
        for(auto pair : map)
            if(pair.second == 2)
                ans ^= pair.first;
        return ans;
    }
};